sycl/source/kernel_bundle.cpp

```diff
@@ -296,28 +290,13 @@ std::vector<kernel_id> get_kernel_ids() {
 }
 
 bool is_compatible(const std::vector<kernel_id> &KernelIDs, const device &Dev) {
-  using namespace detail;
-  std::set<RTDeviceBinaryImage *> BinImages;
-  ProgramManager::getInstance().getRawDeviceImages(KernelIDs, BinImages);
-  for (RTDeviceBinaryImage *Img : BinImages) {
-    const RTDeviceBinaryImage::PropertyRange &PropRange =
-        Img->getDeviceRequirements();
-    for (RTDeviceBinaryImage::PropertyRange::ConstIterator It : PropRange) {
-      using namespace std::literals;
-      if ((*It)->Name != "aspects"sv)
-        continue;
-      ByteArray Aspects = DeviceBinaryProperty(*It).asByteArray();
-      // Drop 8 bytes describing the size of the byte array
-      Aspects.dropBytes(8);
-      while (!Aspects.empty()) {
-        aspect Aspect = Aspects.consume<aspect>();
-        if (!Dev.has(Aspect))
-          return false;
-      }
-    }
-  }
-
-  return true;
+  std::set<sycl::detail::RTDeviceBinaryImage *> BinImages;
+  sycl::detail::ProgramManager::getInstance().getRawDeviceImages(KernelIDs,
+                                                                 BinImages);
```

Comment:
nit: I think we're at least in the `sycl::` namespace already, might strip that part. Not sure about `detail`.